gdnative-core/src/geom/basis.rs

```diff
@@ -7,4 +8,77 @@ pub struct Basis {
     pub elements: [Vector3; 3],
 }
 
-// TODO methods!
+// TODO more methods!
+// Feel free to get inspiration from godot-src\core\math\basis.cpp
+impl Basis {
+    pub fn identity() -> Basis {
+        Basis {
+            elements: [
+                Vector3::new(1.0, 0.0, 0.0),
+                Vector3::new(0.0, 1.0, 0.0),
+                Vector3::new(0.0, 0.0, 1.0),
+            ],
+        }
+    }
+
+    pub fn from_diagonal(p_diag: Vector3) -> Basis {
+        Basis {
+            elements: [
+                Vector3::new(p_diag.x, 0.0, 0.0),
+                Vector3::new(0.0, p_diag.y, 0.0),
+                Vector3::new(0.0, 0.0, p_diag.z),
+            ],
+        }
+    }
+
+    // Note: It's ok to use euclid::UnknownUnit as the type for both Src and Dst
```

Comment:
Maybe this should be in a doc comment instead? I'm pretty sure users will be interested in this information, not just contributors.